### PR TITLE
Associations and add basic UI

### PR DESCRIPTION
**Pull request by kkmanuu**
✅ Create a class Classroom with the following:
⚫ @label instance variable, should be initialized in the constructor.
⚫ Setter and getter for @label (remember about attr_accessor).
      ✅ I Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
    ✅ I Create the has-many side (a classroom has many students).
⚫ Create the belongs-to side (a student belongs to a classroom).
⚫ Make sure that when adding a student to a classroom it also sets the classroom for the student.
⚫ Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
  ✅  I Create a class Book with the following:
⚫ @title and @author instance variables, should be initialized in the constructor.
⚫ Setters and getters for instance variables (remember about attr_accessor).
  ✅ I Create a class Rental with the following:
⚫ @date instance variable, should be initialized in the constructor.
⚫ Setter and getter for @date (remember about attr_accessor).
✅  I Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
⚫ Create the has-many side of Book and Rental (a book has many rentals).
⚫ Create the belongs-to side of Rental and Book (a rental belongs to a book).
⚫ Create the has-many side of Person and Rental (a person has many rentals).
⚫ Create the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @kkmanuu, 

done

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a different milestone activity.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Nemwel-Boniface — COMMENTED** [review]
# Hello @kkmanuu :wave: 

I would like to let you know that we took note of the review that was made for the "OOP school library: add basic UI" milestone by @LionRouge1  and it will be invalidated soon so that when you are done with this milestone "OOP school library: set up associations" and are ready to submit a PR for the "OOP school library: add basic UI" milestone you will be able to do so as a first reviewer and not a re-review.

**However kindly note:**

To avoid similar errors from occurring in future, it would be best to ensure that you are keen when requesting for review so that you do not end up requesting for the review erroneously in the wrong milestone.

Keep rocking and as always #happycoding :rocket:

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
+end
```
Comment (Qoosim) [inline rental.rb]: - [x] Great job setting up associations for your classes. Let's complete the association here. A rental belongs to a book and also belongs to a person. A rental should add itself to both the @book and @person so as to acheive `belong-to` association. It should look something similar to this. Kindly implement that.

```
   def initialize(date, book, person)
      .
      .

      @book = book
      # *The rental is adding itself to the @book achieving the 
       'belongs-to' association
       book.rentals << self

       @person = person
       # *The rental is adding itself to the @person the 'belongs-to' 
       association
       person.rentals << self
    end
```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @kkmanuu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  Implemented project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Fair README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (kkmanuu) [reply rental.rb]: @Qoosim  Done making changes

**Review by skyv26 — APPROVED** [review]
## Hi @kkmanuu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Used correct `GitFlow` workflow ✅ 
- Meaningful **Github** commits ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kkmanuu** [comment]
✔️  I  didn't open another pull request as I was told by my reviewer so in the new commits I:
        ⚫ I Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
⏩ List all books.
⏩List all people.
⏩ I Create a person (teacher or student, not a plain Person).
⏩I Create a book.
⏩I Create a rental.
⏩List all rentals for a given person id.
✔️ I Create main.rb  where  I define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
⏩Present the user with a list of options to perform.
⏩Lets users choose an option.
⏩If needed, ask for parameters for the option.
⏩Have a way to quit the app.

**Review by Gambit142 — COMMENTED** [review]
Hi @kkmanuu 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- Kindly create another branch and another pull request for this milestone. Although reviews were made erroneously previously, you ought to create new branches and new pull request links for each milestone. 👍 😎  

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**